src/pod/utils.ts

```diff
@@ -0,0 +1,75 @@
+import { Metadata, Pod } from './types'
+import { Data } from '@ethersphere/bee-js'
+import { stringToBytes } from '../utils/bytes'
+
+export const metaVersion = 1
+export const MAX_PODS_COUNT = 65536
+
+/**
+ * Extracts pod information from raw data
+ *
+ * @param data raw data with pod information
+ */
+export function extractPods(data: Data): Pod[] {
+  return data
+    .text()
+    .split('\n')
+    .filter(item => Boolean(item.trim()))
+    .map(item => {
+      const parts = item.split(',')
+
+      if (parts.length !== 2) {
+        throw new Error('Pod information: incorrect length')
+      }
+
+      return { name: parts[0], index: Number(parts[1]) } as Pod
+    })
+}
+
+/**
+ * Creates metadata in binary format for pod directory
+ */
+export function createMetadata(
+  version: number,
+  path: string,
+  name: string,
+  creationTime: number,
+  modificationTime: number,
+  accessTime: number,
+): Uint8Array {
+  const data = JSON.stringify({
+    Version: version,
+    Path: path,
+    Name: name,
+    CreationTime: creationTime,
+    ModificationTime: modificationTime,
+    AccessTime: accessTime,
+  } as Metadata)
+
+  return stringToBytes(data)
+}
+
+/**
+ * Verifies if pods list length is correct
+ *
+ * @param length length of pod list
+ */
+export function assertPodsLength(length: number): void {
```

Comment:
I think you should pass a pod list instead because it is required for this context.